app/models/ability.rb

```diff
@@ -15,28 +15,20 @@ def admin_permissions
     can [:manage], :all
   end
 
-  def anonymous_permissions
-    # do not allow viewing incomplete resources
-    curation_concern_read_permissions
-  end
-
-  # Abilities that should be granted to patron
-  def campus_patron_permissions
-    anonymous_permissions
-  end
-
-  def completer_permissions
-    can [:read, :modify, :update], curation_concerns
-    can [:read, :edit, :update], FileSet
-    can [:read, :edit, :update], Collection
-
-    # allow completing resources
-    can [:complete], curation_concerns
-
-    curation_concern_read_permissions
+  # Staff can do anything except delete someone else's stuff
+  def staff_permissions
+    can [:create, :read, :modify, :update], :all
+    can [:destroy], Template
+    can [:destroy], FileSet do |obj|
+      obj.depositor == [current_user.uid]
+    end
+    can [:destroy], curation_concerns do |obj|
+      obj.depositor == [current_user.uid]
+    end
   end
 
-  def curation_concern_read_permissions
+  def anonymous_permissions
+    # do not allow viewing incomplete resources
     cannot [:read], curation_concerns do |resource|
```

Comment:
Can you reverse the logic of these two `cannot` blocks to be `can :read readable_concern?` etc.? I think that would be easier to read, and avoid the mixing of `can` and `cannot` which we thought might cause problems.

Also, it looks like `anonymous_permissions` is only called from the `campus_patron_permissions` below — maybe we should just fold this into that method?

Reply:
`anonymous_permissions` is also called dynamically in `custom_permissions`

I will look into reversing the `cannot` logic.